src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -537,7 +539,38 @@ public unsafe int Seek(ref Vector<byte> byte0Vector, ref Vector<byte> byte1Vecto
             }
         }
 
-        private static int FindFirstEqualByte(ref Vector<byte> byteEquals)
+        // Internal for testing
+        /// <summary>
+        /// Find first byte
+        /// </summary>
+        /// <param  name="byteEquals"></param >
+        /// <returns>The first index of the result vector</returns>
+        /// <exception cref="InvalidOperationException">byteEquals = 0</exception>
+        internal static int FindFirstEqualByte(ref Vector<byte> byteEquals)
```

Comment:
I'm not very fond of the idea of putting "for testing" code outside of test code. Why does this have to here?